android/java/org/chromium/chrome/browser/vpn/split_tunnel/ApplicationListAdapter.java

```diff
@@ -0,0 +1,152 @@
+                        notifyItemRangeChanged(position, count);
+                    }
+
+                    @Override
+                    public boolean areContentsTheSame(
+                            ApplicationDataModel oldItem, ApplicationDataModel newItem) {
+                        return oldItem.getName().equals(newItem.getName());
+                    }
+
+                    @Override
+                    public boolean areItemsTheSame(
+                            ApplicationDataModel item1, ApplicationDataModel item2) {
+                        return item1.getName().equals(item2.getName());
+                    }
+
+                    @Override
+                    public void onInserted(int position, int count) {
+                        notifyItemRangeInserted(position, count);
+                    }
+
+                    @Override
+                    public void onRemoved(int position, int count) {
+                        notifyItemRangeRemoved(position, count);
+                    }
+
+                    @Override
+                    public void onMoved(int fromPosition, int toPosition) {
+                        notifyItemMoved(fromPosition, toPosition);
+                    }
+                });
+        this.mOnApplicationClickListener = onApplicationClickListener;
+        this.mIsExcludedApps = isExcludedApps;
+    }
+
+    public void addAll(List<ApplicationDataModel> items) {
+        mApplicationList.beginBatchedUpdates();
+        for (ApplicationDataModel item : items) {
+            mApplicationList.add(item);
+        }
+        mApplicationList.endBatchedUpdates();
+    }
+
+    @NonNull
+    @Override
+    public RecyclerView.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
+        View rootView = LayoutInflater.from(parent.getContext())
+                                .inflate(R.layout.application_item_layout, parent, false);
+        return new ApplicationDataViewHolder(rootView);
+    }
+
+    @Override
+    public void onBindViewHolder(@NonNull RecyclerView.ViewHolder holder, int position) {
+        ApplicationDataModel applicationDataModel = mApplicationList.get(position);
+        ApplicationDataViewHolder viewHolder = (ApplicationDataViewHolder) holder;
+
+        viewHolder.mAppName.setText(applicationDataModel.getName());
+        viewHolder.mAppIcon.setImageDrawable(applicationDataModel.getIcon());
+        viewHolder.mActionIcon.setImageResource(
+                mIsExcludedApps ? R.drawable.ic_baseline_close_24 : R.drawable.ic_baseline_add_24);
+        viewHolder.mActionIcon.setOnClickListener(view
+                -> mOnApplicationClickListener.onApplicationCLick(
+                        applicationDataModel, position, mIsExcludedApps));
+        setFadeAnimation(viewHolder.itemView);
+    }
+
+    private void setFadeAnimation(View view) {
+        AlphaAnimation anim = new AlphaAnimation(0.0f, 1.0f);
+        anim.setDuration(1000);
```

Comment:
It's better to use constants instead of hardcoding numbers for easier adjustments in future, if necessary